components/transfer/Transfer.vue

```diff
@@ -495,44 +530,71 @@ const handleOpenConfirmModal = () => {
   }
 }
 
-const getAmountToTransfer = (amount: number, decimals: number) =>
-  String(calculateBalance(Number(amount), decimals))
+const getTransactionFee = async () => {
+  const [cb, params] = await getTransferParmas(
+    targetAddresses.value.map(
+      () =>
+        ({
+          address: toDefaultAddress(KODADOT_DAO),
+          usd: 1,
+          token: 1,
+        } as TargetAddress)
+    ),
+    decimals.value as number
+  )
 
-interface TransferParams {
-  api: ApiPromise
-  decimals: number
+  return estimate(accountId.value, cb as any, params as any)
 }
 
-const getMultipleAddressesTransferParams = ({
-  api,
-  decimals,
-}: TransferParams) => {
-  const arg = [
-    targetAddresses.value.map((target) => {
-      const amountToTransfer = getAmountToTransfer(
-        target.token as number,
-        decimals
-      )
+const calculateTransactionFee = async () => {
+  txFee.value = 0
+  const fee = await getTransactionFee()
+  txFee.value = Number((Number(fee) / Math.pow(10, decimals.value)).toFixed(4))
+}
 
-      return api.tx.balances.transfer(
-        target.address as string,
-        amountToTransfer
-      )
-    }),
-  ]
+onMounted(() => calculateTransactionFee())
 
-  return [api.tx.utility.batch, arg]
-}
+watchDebounced(
+  [urlPrefix, () => targetAddresses.value.length],
+  () => {
+    calculateTransactionFee()
+  },
+  { debounce: 500 }
+)
 
-const getSingleAddressTransferParams = ({ api, decimals }: TransferParams) => {
-  const target = targetAddresses.value[0]
+const getAmountToTransfer = (amount: number, decimals: number) =>
+  String(calculateBalance(Number(amount), decimals))
 
-  const amountToTransfer = getAmountToTransfer(target.token as number, decimals)
+const getTransferParmas = async (
+  addresses: TargetAddress[],
+  decimals: number
+) => {
+  const api = await apiInstance.value
+  const isSingle = targetAddresses.value.length === 1
+
+  const firstAddress = addresses[0]
+
+  const cb = isSingle ? api.tx.balances.transfer : api.tx.utility.batch
```

Comment:
could be further simplified by deciding on the callback in `submit` function
making `getTransferParmas` function return only the args and not an array

Reply:
I'd leave it like that otherwise ill have to decide cb also inside `getTransactionFee` , having a little bit of duplication
<img width="640" alt="CleanShot 2023-08-30 at 10 32 58@2x" src="https://github.com/kodadot/nft-gallery/assets/44554284/f5b19f9b-c28a-42ef-a725-fb63ab44bb0b">